demo.py

```diff
@@ -121,13 +121,14 @@ def build_cqm(G, k):
 
     # Add binary variables, one for each node and each partition in the graph
     print("\nAdding variables....")
-    v = [[Binary(f'v_{i},{k}') for k in partitions] for i in G.nodes]   
+    v = [[Binary(f'v_{i},{k}') for k in partitions] for i in G.nodes]
+    labels = [[f'v_{i},{k}' for k in partitions] for i in G.nodes]
 
     # One-hot constraint: each node is assigned to exactly one partition
     print("\nAdding one-hot constraints...")
     for n in G.nodes:
         # print("\nAdding one-hot for node", n)
-        cqm.add_constraint(quicksum(v[n]) == 1, label='one-hot-node-{}'.format(n)) 
+        cqm.add_discrete(labels[n], label=f"one-hot-node-{n}")
```

Comment:
Alternatively, you could could combine this line with line 125:
```
for n in G.nodes:
  cqm.add_discrete([f'v_{n},{k}' for k in partitions], label=f"one-hot-node-{n}")
```

Reply:
Thanks for the suggestion! I used it instead of what I had :)